### PR TITLE
guides: deployment

### PR DESCRIPTION
**Pull request by ElysaSrc**
This modification add guides that cover deployments

- [X] Describe container architecture
  - [X] English
  - [X] French
- [X] Helm depoloyments
  - [X] English
  - [X] French
- [X] Docker compose deployments
  - [X] English
  - [X] French

### CODE REVIEW

**Review by multun — APPROVED** [review]

File: content/docs/explanation/containers-architecture/_index.fr.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+weight: 10
+description: "Comment les conteneurs fonctionnent ensemble et comment ils sont construits"
+---
+
+Il y a 3 principaux conteneurs déployés dans une configuration OSRD standard :
+ - **Gateway** _(inclut le frontend)_ : Sert le front-end, gère l'authentification et proxy les requêtes vers le backend.
+ - **Editoast** : Agit comme le backend qui interagit avec le front-end.
+ - **Core** : Gère les calculs et la logique métier, appelé par Editoast.
+
+## Standard deployment
```
Comment (celine-sncf) [inline content/docs/explanation/containers-architecture/_index.fr.md:13]:
```suggestion
## Déploiement standard
```

File: content/docs/explanation/containers-architecture/_index.fr.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    gw -- fichier local --> front
+    
+    navigateur --> gw
+    gw -- HTTP --> editoast
+    editoast -- HTTP --> core
+```
+
+Les requêtes externes sont reçues par le gateway. Si le chemin demandé commence par `/api`, il sera transféré en utilisant HTTP vers editoast, sinon il servira un fichier avec le chemin demandé. Editoast atteint le core en utilisant HTTP si nécessaire.
+
+Le gateway n'est pas seulement un proxy inverse avec le bundle front-end inclus, il fournit également tous les mécanismes d'authentification : utilisant OIDC ou des jetons.
```
Comment (celine-sncf) [inline content/docs/explanation/containers-architecture/_index.fr.md:30]:
```suggestion
Le gateway n'est pas seulement un proxy inverse avec le bundle front-end inclus, il fournit également tous les mécanismes d'authentification : utilisant OIDC ou des tokens.
```

File: content/docs/guides/deploy/_index.en.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+First of all, we recommend learning about the [containers architecture of OSRD]({{< ref "/docs/explanation/containers-architecture" >}}).
+
+We will cover how to deploy OSRD within the following setups:
+
+ - [Using docker compose]({{< ref "/docs/guides/deploy/docker-compose" >}}) on a single node.
+ - [Using helm]({{< ref "/docs/guides/deploy/docker-compose" >}}) on a kubernetes cluster.
+
+It's also possible to deploy manually each service of OSRD manually on a system, but we will not cover this topic within this guide.
```
Comment (celine-sncf) [inline content/docs/guides/deploy/_index.en.md:15]:
```suggestion
It is also possible to deploy each service of OSRD manually on a system, but we will not cover this topic within this guide.
```

File: content/docs/guides/deploy/docker-compose/_index.en.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+1. **PostgreSQL**: A PostgreSQL database with PostGIS extension. 
+2. **Redis**: A Redis server for caching.
+3. **Core**: The core OSRD service.
+4. **Front**: The front-end service for OSRD.
+5. **Editoast**: A service responsible for various editorial functions.
+6. **Gateway**: Serves as the gateway for the OSRD services.
+7. **Wait-Healthy**: A utility service to ensure all services are healthy before proceeding.
+
+Each service is configured with health checks, volume mounts, and necessary environment variables.
```
Comment (celine-sncf) [inline content/docs/guides/deploy/docker-compose/_index.en.md:28]:
```suggestion
Each service is configured with health checks, volume mounts and necessary environment variables.
```

**Review by celine-sncf — CHANGES_REQUESTED** [review]
corrected a few typos

File: content/docs/guides/deploy/kubernetes/_index.fr.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    gw -- HTTP --> editoast
+    gw -- HTTP --> tileserver-1
+    gw -- HTTP --> tileserver-2
+    gw -- HTTP --> tileserver-n...
+    editoast -- HTTP --> core
+```
+
+Le Helm Chart utilise le[HorizontalPodAutoscaler](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale/) de Kubernetes pour lancer autant de tileserver que nécessaire en fonction de la charge de travail.
+
+## Configuration de la Helm Chart (values)
```
[the remark this answers is not in the document]
Comment (ElysaSrc) [reply content/docs/guides/deploy/kubernetes/_index.fr.md:40]: Ici je laisse values car c'est le terme technique des helm charts :)

**Review by celine-sncf — APPROVED** [review]
thank you for the corrections :)

**Review by multun — APPROVED** [review]

**Review by multun — APPROVED** [review]

**Pull request merged**